core/services/assets/BlockAssetManager.php

```diff
@@ -0,0 +1,316 @@
+        }
+        return parent::addJavascript(
+            $handle,
+            $this->registry->getJsUrl(
+                $this->domain->assetNamespace(),
+                $handle
+            ),
+            $this->addDefaultBlockScriptDependencies($dependencies)
+        );
+    }
+
+
+    /**
+     * @param        $handle
+     * @param array  $dependencies
+     * @since $VID:$
+     * @return StylesheetAsset
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     * @throws DuplicateCollectionIdentifierException
+     */
+    public function addStyle($handle, array $dependencies = array())
+    {
+        if ($this->assets->has($handle)) {
+            return $this->assets->get($handle);
+        }
+        return parent::addStylesheet(
+            $handle,
+            $this->registry->getCssUrl(
+                $this->domain->assetNamespace(),
+                $handle
+            ),
+            $dependencies
+        );
+    }
+
+
+    /**
+     * @param array $dependencies
+     * @return array
+     */
+    protected function addDefaultBlockScriptDependencies(array $dependencies)
+    {
+        $dependencies += array(
+                CoreAssetManager::JS_HANDLE_EE_JS_CORE,
+                'wp-blocks',    // Provides useful functions and components for extending the editor
+                'wp-i18n',      // Provides localization functions
+                'wp-element',   // Provides React.Component
+                'wp-components' // Provides many prebuilt components and controls
+            );
+        return $dependencies;
+    }
+
+
+    /**
+     * @param string $handle
+     * @since $VID:$
+     * @return mixed|null
+     */
+    public function getAsset($handle)
+    {
+        if ($this->assets->has($handle)) {
+            return $this->assets->get($handle);
+        }
+        return null;
+    }
+
+
+    /**
+     * @return JavascriptAsset|null
+     */
+    public function getEditorScript()
+    {
+        return $this->getAsset($this->editor_script_handle);
+    }
+
+
+    /**
+     * @return StylesheetAsset|null
+     */
+    public function getEditorStyle()
+    {
+        return $this->getAsset($this->editor_style_handle);
+    }
+
+
+    /**
+     * @return JavascriptAsset|null
+     */
+    public function getScript()
+    {
+        return $this->getAsset($this->script_handle);
+    }
+
+
+    /**
+     * @return StylesheetAsset|null
+     */
+    public function getStyle()
+    {
+        return $this->getAsset($this->style_handle);
+    }
+
+
+    /**
+     * @return  void
+     */
+    public function enqueueAssets()
+    {
+        $assets = array(
+            $this->getEditorScript(),
+            $this->getEditorStyle(),
+            $this->getScript(),
+            $this->getStyle(),
+        );
+        foreach ($assets as $asset) {
+            if ($asset instanceof BrowserAsset && $asset->isRegistered()) {
+                $asset->enqueueAsset();
```

Comment:
This may be already addressed, but just want to make sure that if blocks have been registered for only specific cpts (i.e. posts or pages and NOT EE Cpts), are their assets still only going to be _enqueued_ for those pages?  This is a bit related to my question about the purpose for `core-blocks` other words, is `core-blocks` going to be for blocks that are available on any editor?

Reply:
`CoreBlocksAssetManager` is basically all of the asset configuration that used to be directly within `core\services\assets\Registry`
so it would have run regardless of the CPT type. 
The `Asset::isRegistered()` method is just to prevent assets from being registered multiple times and has no correlation to CPT type or whether a script gets added for a particular post type. Some of that logic may still need to be figured out.

Reply:
My comment was less related to the `$asset->isRegistered()` bit and more related to the `$asset->enqueueAsset()` bit here.   Registration can happen on any route, but enqueueing should only be done where the script is actually needed.  In the case of blocks this is important because although there is a server side php registration, if a script is enqueued, then the block is also registered.  Which comes back to the latter part of my question which is, "is `core-blocks` going to be for blocks that are available in any editor?

Reply:
The other part of this is that the WP Gutenberg block registration php side _should_ automatically take _care_ of _enqueueing_ assets for the block.  Since that's the case, why is this even needed here?